### PR TITLE
Enable BatchNorm for Keras Model in DDPG Example

### PR DESCRIPTION
**Pull request by amifunny**
`training=True` was added to each model call to enable BatchNorm.

### CODE REVIEW

File: examples/rl/ddpg_pendulum.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def get_critic():
 
 
 def policy(state, noise_object):
-    sampled_actions = tf.squeeze(actor_model(state))
+    sampled_actions = tf.squeeze(actor_model(state,training=True))
```
Comment (fchollet) [inline examples/rl/ddpg_pendulum.py:296]: I think this function should run in inference mode (training=False) -- no?

File: examples/rl/ddpg_pendulum.py [hunk 2]
```diff
@@ -187,9 +187,9 @@ def learn(self):
         # Training and updating Actor & Critic networks.
         # See Pseudo Code.
         with tf.GradientTape() as tape:
-            target_actions = target_actor(next_state_batch)
-            y = reward_batch + gamma * target_critic([next_state_batch, target_actions])
-            critic_value = critic_model([state_batch, action_batch])
+            target_actions = target_actor(next_state_batch,training=True)
```
Comment (fchollet) [inline examples/rl/ddpg_pendulum.py:190]: Please use `black` to format the code (space after comma)

File: examples/rl/ddpg_pendulum.py [hunk 3]
```diff
@@ -12,11 +12,13 @@
...
 It combines ideas from DPG (Deterministic Policy Gradient) and DQN (Deep Q-Network).
-It uses Experience Replay and slow-learning target networks from DQN, and it is based on DPG,
+It uses Experience Replay and slow-learning target networks from DQN, and it is based on
+DPG,
 which can operate over continuous action spaces.
 
 This tutorial closely follow this paper -
-[Continuous control with deep reinforcement learning](https://arxiv.org/pdf/1509.02971.pdf)
+[Continuous control with deep reinforcement
+learning](https://arxiv.org/pdf/1509.02971.pdf)
```
Comment (fchollet) [inline examples/rl/ddpg_pendulum.py:21]: Put the link a single line so it renders properly.

File: examples/rl/ddpg_pendulum.py [hunk 4]
```diff
@@ -414,14 +435,17 @@ def policy(state, noise_object):
...
 
 ![before_img](https://i.imgur.com/ox6b9rC.gif)
+
 """
 
 """
 After 100 episodes:
 
 ![after_img](https://i.imgur.com/eEH8Cz6.gif)
+
```
Comment (fchollet) [inline examples/rl/ddpg_pendulum.py:450]: How did the extra whitespace get added? You can remove it.

Comment (amifunny) [reply examples/rl/ddpg_pendulum.py:449]: Yeah, Many blank lines are added by `black`.

Comment (fchollet) [reply examples/rl/ddpg_pendulum.py:449]: `black` does not add any such blank lines. Let's get rid of them.

- Start from the original file (on the `master` branch)
- Add the `training=True` statements and modify the "last modified" date
- Run `black ddpg_pendulum.py`
- Commit the file

**Review by fchollet — COMMENTED** [review]
You can also just remove the added extra space by hand...

Please add the generated files.

**Comment by amifunny** [comment]
Sorry for the late response. Deleted extra lines manually. 

Generating files with cmd -
`python autogen.py add_example rl/ddpg_pendulum`

gave the following error -
`Traceback (most recent call last):
File "../scripts/autogen.py", line 918, in <module>
working_dir=get_working_dir(sys.argv[3]) if len(sys.argv) == 4 else None,
File "../scripts/autogen.py", line 212, in add_example
assert path.count(os.path.sep) == 1
AssertionError
`
Can you add it on your part?

**Review by fchollet — APPROVED** [review]
Thanks, I will refresh the files then.

**Comment by fchollet** [comment]
FYI batchnorm breaks training entirely so I have removed it.

**Pull request merged**